### PR TITLE
Feature/hotspot settings

### PR DESCRIPTION
**Pull request by matthewcarlreetz**
On hotspot detail when pressing the hamburger menu an action sheet will show. Pressing the "settings" option will launch the diagnostic flow.

https://user-images.githubusercontent.com/1760821/105385440-f962d700-5bd8-11eb-8fdf-cc20fe4461cf.mov

### CODE REVIEW

File: src/features/hotspots/details/HotspotDetailCardHeader.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ const HotspotDetailCardHeader = ({
         <Heart />
       </TouchableOpacityBox>
       <CardHandle />
-      <TouchableOpacityBox onPress={onMoreSelected}>
+      <TouchableOpacityBox onPress={onMoreSelected} padding="m" margin="n_m">
```
Comment (allenan) [inline src/features/hotspots/details/HotspotDetailCardHeader.tsx:23]: just curious what the padding + negative margin does here?

File: src/features/hotspots/details/HotspotDetails.tsx [hunk 2]
```diff
@@ -59,6 +57,31 @@ const HotspotDetails = () => {
   const { t } = useTranslation()
   const selectedHotspots = hotspotsToFeatures([hotspot])
   const { purpleMain, greenOnline } = useColors()
+  const { showActionSheetWithOptions } = useActionSheet()
+  const [showSettings, setShowSettings] = useState(false)
+
+  const onMoreMenuSelected = () => {
+    const options: string[] = [
+      ...t('hotspot_details.more', { returnObjects: true }),
```
Comment (allenan) [inline src/features/hotspots/details/HotspotDetails.tsx:65]: returning an array here, and then making assumptions below based on the index feels a little implicit. would it be worth having this be a bit more explicit at the cost of being more verbose?

Comment (allenan) [reply src/features/hotspots/details/HotspotDetails.tsx:65]: you could maybe create an object with labels and handler functions. would remove the need for the switch below, potentially

File: src/features/hotspots/root/HotspotsView.tsx [hunk 3]
```diff
@@ -178,27 +175,15 @@ const HotspotsView = ({ ownedHotspots }: Props) => {
...
-          >
-            <Settings />
-          </TouchableOpacityBox>
-        )}
         initialSnapIndex={1}
         snapProgress={snapProgress}
       >
         <HotspotsCarousel
-          hotspots={ownedHotspots}
+          hotspots={[...ownedHotspots, ...ownedHotspots]}
```
Comment (allenan) [inline src/features/hotspots/root/HotspotsView.tsx:182]: is this maybe left in from testing?

File: src/features/hotspots/details/HotspotDetailCardHeader.tsx (hunk 1 above)
Comment (matthewcarlreetz) [reply src/features/hotspots/details/HotspotDetailCardHeader.tsx:23]: It gives some hit slop. The button needed some padding, but the container had the padding already. This gives it the padding, then the n_m repositions it where it should be.

File: src/features/hotspots/details/HotspotDetails.tsx (hunk 2 above)
Comment (matthewcarlreetz) [reply src/features/hotspots/details/HotspotDetails.tsx:65]: yeah, I'm with you. It does feel a bit smelly.

File: src/features/hotspots/root/HotspotsView.tsx (hunk 3 above)
Comment (matthewcarlreetz) [reply src/features/hotspots/root/HotspotsView.tsx:182]: damn. Yes.

File: src/features/hotspots/details/HotspotDetails.tsx (hunk 2 above)
Comment (matthewcarlreetz) [reply src/features/hotspots/details/HotspotDetails.tsx:65]: Your suggestion is similar to what I did for the more page. I'll go with that.

**Review by tyler-whitman — APPROVED** [review]
Looks good, just wanted to make sure the yarn changes are ok? Looks like a lot changed

**Review by allenan — APPROVED** [review]
awesome, send it!

**Review by DGKSK8LIFE — APPROVED** [review]
Lookin' good!

**Pull request merged**